### PR TITLE
feat: adding relevant files for PPO

### PR DESCRIPTION
**Pull request by Nilay27**
* Adding relevant files for PPO contract
* Also added overwritten function signature of `IPPO.sol` to avoid making the contract abstract as all those methods were already defined on `IPPO.sol` beforehand.

### CODE REVIEW

File: apps/smart-contracts/token/contracts/PPO.sol [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+contract PPO is IPPO, ReentrancyGuardUpgradeable, SafeOwnableUpgradeable {
+  function initialize(address _nominatedOwner) public initializer {}
+
+  function setTransferHook(ITransferHook _newTransferHook) external override onlyOwner {}
+
+  function mint(uint256 _amount) external override onlyOwner {}
+
+  function burn(uint256 _amount) external override onlyOwner {}
+
+  function burnFrom(address _account, uint256 _amount) external override onlyOwner {}
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/contracts/PPO.sol:17]: these are not onlyOwner functions

File: apps/smart-contracts/token/contracts/PPO.sol [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity =0.8.7;
+
+import "./interfaces/IPPO.sol";
+import "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import "prepo-shared-contracts/contracts/SafeOwnableUpgradeable.sol";
+
+contract PPO is IPPO, ReentrancyGuardUpgradeable, SafeOwnableUpgradeable {
```
Comment (XavierEkkel) [inline apps/smart-contracts/token/contracts/PPO.sol:8]: reentrancy guard isn't used anywhere

**Review by XavierEkkel — APPROVED** [review]
LGTM after comments addressed

File: apps/smart-contracts/token/test/PPO.test.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import chai from 'chai'
+import { solidity } from 'ethereum-waffle'
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/token/test/PPO.test.ts:2]:
```suggestion
```
Remember, we don't use `ethereum-waffle` anymore, and use the `@nomiclabs/hardhat-chai-matchers` library.

This is automatically imported in ` hardhat.config.ts` and does not need to be reimported.

File: apps/smart-contracts/token/test/PPO.test.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
...
+import { ppoFixture } from './fixtures/PPOFixtures'
+
+chai.use(solidity)
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/token/test/PPO.test.ts:5]:
```suggestion
```
Remove this as well, no longer need this due to the above.

**Review by ramenforbreakfast — APPROVED** [review]
LGTM besides Xavier's comments and my nits.

**Pull request merged**